src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupBy = String(item[key]);
+
+    groupedItems[groupBy] = items.filter(
+      (element) => String(element[key]) === groupBy,
+    );
```

Comment:
Not an optimal way. You will loop multiple times over a whole array in case of duplicate keys. Better add a check if `groupBy` prop is already present in the object to avoid an unnecessary array loop.